### ISSUE TITLE
508-defect-2 [KEYBOARD]: VAMC Templates - On this page links must set focus correctly in iOS Safari + VoiceOver

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#21067]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact
**VFS Point of Contact:** _Angela or Trevor_

## User Story or Problem Statement
As an iOS smartphone user who also runs VoiceOver, I want focus to be managed correctly when I double-click on on On This Page link.

## Details
While testing the Eastern Oklahoma VA Medical Center with iOS Safari + VoiceOver, I noticed the On This Page links do not set focus on the deep-linked heading. The page jumps to the correct location, but when I swipe right to move to the next text element, the page jumps back to the next On This Page link.

## Acceptance Criteria
- [ ] iOS Safari + VoiceOver manages focus correctly
- [ ] When users double-tap an On This Page link, the heading receives focus
- [ ] When users swipe right after the heading is read, their virtual cursor should be on the next text element and the page should not jump back to On This Page.

## Environment
* https://www.va.gov/eastern-oklahoma-health-care/health-services/ (example page, this appears to be a template issue)

## Steps to Recreate

1. Enter `<IP ADDRESS>` in browser
2. Turn on iOS VoiceOver, switch to Safari browser
3. Swipe right to the first On This Page link, "Primary Care"
4. Double-tab the Primary Care link to jump to Primary Care H2
5. Swipe right. Your focus *should* be on the Emergency Care accordion, but it will likely jump back up to Mental Health Care link.

## Proposed Solution (if known)
Safari on MacOS (desktop) used to require a `tabindex="-1"` on the headings that are used as our anchor targets.

```diff
<h2
  id="primary-care"
+ tabindex="-1"
>Primary care</h2>
```

## WCAG or Vendor Guidance (optional)
* [Focus Order: Understanding SC 2.4.3](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-focus-order.html)

**Comment by jilladams** [comment]
@laflannery at your leisure, mind verifying if this is "still applicable"?

**Comment by laflannery** [comment]
@jilladams This specific issue looks to have been resolved but I am encountering a different one:

- Same set of circumstances (mobile, safari, VoiceOver, VAMC page template, On this page section)
- If I try to use any of the other links (not the first Primary Care link) the focus jumps to the last accordion in the previous section, instead of the  heading the corresponds with the link I'm trying to go to.  The next tab stop is the heading but this is still confusing for the user because they should be hearing the heading immediately

Do you want me to close this ticket and open a new ticket in the CMS repo for this new focus issue I discovered?

**Comment by jilladams** [comment]
Yes please! Closing this.

**Comment by laflannery** [comment]
Done: https://github.com/department-of-veterans-affairs/va.gov-cms/issues/13946
